pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/AvgAggregationFunction.java

```diff
@@ -60,12 +62,34 @@ public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int ma
   @Override
   public void aggregate(int length, @Nonnull AggregationResultHolder aggregationResultHolder,
       @Nonnull BlockValSet... blockValSets) {
-    double[] valueArray = blockValSets[0].getDoubleValuesSV();
-    double sum = 0.0;
-    for (int i = 0; i < length; i++) {
-      sum += valueArray[i];
+    FieldSpec.DataType valueType = blockValSets[0].getValueType();
+    switch (valueType) {
+      case INT:
+      case LONG:
+      case FLOAT:
+      case DOUBLE:
+        double[] valueArray = blockValSets[0].getDoubleValuesSV();
+        double sum = 0.0;
+        for (int i = 0; i < length; i++) {
+          sum += valueArray[i];
+        }
+        setAggregationResult(aggregationResultHolder, sum, (long) length);
+        break;
+      case BYTES:
+        // Serialized AvgPair
+        byte[][] bytesValues = blockValSets[0].getBytesValuesSV();
+        sum = 0.0;
+        long count = 0L;
+        for (int i = 0; i < length; i++) {
+          AvgPair value = ObjectSerDeUtils.AVG_PAIR_SER_DE.deserialize(bytesValues[i]);
```

Comment:
How do we guarantee that the bytes are deserializable to the expected class?

Reply:
The bytes should be generated from star-tree (or the aggregator for this aggregation function). If client aggregates on a random bytes column, it will throw exception.

Reply:
For the latter, we probably should think about having a query sanitizer to block bad queries reaching this deep in the execution stack. But that I guess is outside the scope of this PR.